test/functional/helper.go

```diff
@@ -245,6 +262,29 @@ func GetServiceDetailsCosmos(servicesJson string, service string) (endpoint stri
 
 }
 
+func GetServiceDetailsIcon(servicesJson string, service string) (endpoint string) {
+	var data map[string]CosmosNodeInfo
```

Comment:
You are using cosmos node info here not icon and if getting the endpoint logic is same for both cosmos and icon why don't you make it generic ?